syncapi/storage/interface.go

```diff
@@ -104,4 +105,16 @@ type Database interface {
 	StreamEventsToEvents(device *authtypes.Device, in []types.StreamEvent) []gomatrixserverlib.HeaderedEvent
 	// SyncStreamPosition returns the latest position in the sync stream. Returns 0 if there are no events yet.
 	SyncStreamPosition(ctx context.Context) (types.StreamPosition, error)
+	// AddSendToDevice increases the EDU position in the cache and returns the stream position.
+	AddSendToDevice() types.StreamPosition
```

Comment:
It seems like a weird API shape to have this decoupled from `StoreNewSendForDeviceMessage` which would surely increment the stream position..? Why does it have this shape?